modules/spotxBidAdapter.js

```diff
@@ -0,0 +1,395 @@
+import * as utils from 'src/utils';
+import { Renderer } from 'src/Renderer';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { VIDEO } from 'src/mediaTypes';
+
+const BIDDER_CODE = 'spotx';
+const URL = '//search.spotxchange.com/openrtb/2.3/dados/';
+const ORTB_VERSION = '2.3';
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['spotx'],
+  supportedMediaTypes: [VIDEO],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   * From Prebid.js: isBidRequestValid - Verify the the AdUnits.bids, respond with true (valid) or false (invalid).
+   *
+   * @param {object} bid The bid to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function(bid) {
+    if (bid && typeof bid.params !== 'object') {
+      utils.logError(BIDDER_CODE + ': params is not defined or is incorrect in the bidder settings.');
+      return false;
+    }
+
+    if (!utils.deepAccess(bid, 'mediaTypes.video')) {
+      utils.logError(BIDDER_CODE + ': mediaTypes.video is not present in the bidder settings.');
+      return false;
+    }
+
+    const playerSize = utils.deepAccess(bid, 'mediaTypes.video.playerSize');
+    if (!playerSize || !utils.isArray(playerSize)) {
+      utils.logError(BIDDER_CODE + ': mediaTypes.video.playerSize is not defined in the bidder settings.');
+      return false;
+    }
+
+    if (!utils.getBidIdParameter('channel_id', bid.params)) {
+      utils.logError(BIDDER_CODE + ': channel_id is not present in bidder params');
+      return false;
+    }
+
+    const videoMediaType = utils.deepAccess(bid, 'mediaTypes.video');
+    const context = utils.deepAccess(bid, 'mediaTypes.video.context');
+    if ((bid.mediaType === 'video' || (videoMediaType && context === 'outstream')) || utils.getBidIdParameter('ad_unit', bid.params) == 'outstream') {
+      if (!utils.getBidIdParameter('outstream_function', bid.params)) {
+        if (!utils.getBidIdParameter('outstream_options', bid.params)) {
+          utils.logError(BIDDER_CODE + ': please define outstream_options parameter or override the default SpotX outstream rendering by defining your own Outstream function using field outstream_function.');
+          return false;
+        }
+        if (!utils.getBidIdParameter('slot', bid.params.outstream_options)) {
```

Comment:
In the .md file, this param appears to be named `video_slot` instead of just `slot`.  Which is the correct variable name?

Reply:
You are right, it's slot and not video_slot. We will change it in the md file.